hail/src/main/java/is/hail/io/compress/BGzipInputStream.java

```diff
@@ -202,9 +215,6 @@ public int read(byte[] b, int off, int len) throws IOException {
         System.arraycopy(outputBuffer, outputBufferPos, b, off, toCopy);
         outputBufferPos += toCopy;
 
-        if (outputBufferPos == outputBufferSize)
```

Comment:
I don't quite get why you're making this change.  If you leave this here, then the internal state satisfies the external contract, and you don't need a test in `getVirtualOffset` and `inputBufferInPos` is the current block pos so you don't need `currentBlockPos`.

Reply:
I wasn't convinced of that. I am now, given that the tests pass. I originally made that change for a small efficiency gain; if the call stopped reading, no more data would be decompressed. This particular bit of complexity would have been hidden and I got it correct, but :man_shrugging: